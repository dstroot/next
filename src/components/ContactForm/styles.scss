@import '../../scss/main.scss';

@mixin box-shadow($val) {
  -webkit-box-shadow: $val;
  -moz-box-shadow: $val;
  -ms-box-shadow: $val;
  box-shadow: $val;
}

label {
  margin-bottom: 0;
}

.alert-success {
  font-family: $circular;
  font-size: calc(18px);
  text-align: center;
  padding: 10px;
  background-color: theme-color('secondary-dark');
  border-left: 12px solid theme-color('primary');
  display: none;
  color: theme-color('primary-dark');
}

.alert-fail {
  font-family: $circular;
  font-size: calc(18px);
  text-align: center;
  padding: 10px;
  background-color: theme-color('secondary-dark');
  border-left: 12px solid #f44336;
  display: none;
}

.form-control:focus {
  border-color: theme-color('primary');
  @include box-shadow(none);
}

input:invalid {
  border-color: none;
}

.input-fail {
  text-align: center;
  display: block;
}

.form-input-fail {
  border: 1px solid #f44336;
}
