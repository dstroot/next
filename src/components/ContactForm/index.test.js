import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import ContactForm from '../ContactForm';

describe('ContactForm', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<ContactForm />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<ContactForm />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
