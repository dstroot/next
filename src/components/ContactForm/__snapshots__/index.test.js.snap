// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm it should render 1`] = `
<div
  className="text-left nxt_body-xsmall"
>
  <form
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="react-reveal alert-success nxt_body-small"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <p>
        We received your message! Weâ€™ll get back to you shortly.
      </p>
      <p>
        If this is an urgent matter, pleae call us at (833) 646-6398
      </p>
    </div>
    <div
      className="react-reveal alert-fail nxt_body-small"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <span
        className="input-fail"
        id="fail-message"
      />
      <span
        className="input-fail"
        id="invalid-email-message"
      />
    </div>
    <div>
      <label>
        Name
      </label>
      <input
        className="form-control form-input"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Email
      </label>
      <input
        className="form-control form-input"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Phone Number
      </label>
      <input
        autoComplete="tel"
        className="form-control mb-3"
        maxLength="14"
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        type="tel"
        value=""
      />
    </div>
    <div>
      <label>
        Message
      </label>
      <textarea
        className="form-control mb-3 form-input"
        name="message"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-primary nxt_rounded-pill nxt_body-small px-4 py-2 submit"
    >
      Submit
    </button>
  </form>
</div>
`;
